### ISSUE TITLE
Program exits while listing all music / artists

### ISSUE THREAD

**Issue by snz138** [akshay2000/XBMCRemoteWP#5]
1) I open X.Assist and It connects
2) I go to "music"
3) Select "all music"
4) "artists" view opens up and it reads "Loading..." on top left. Five dots on top of the screen keep flying from left to right for about a minute and then the program exits.

I use XBMC (Gotham 13.2) and it lists 2837 artists. Some of them are in Japanese characters.

**Comment by akshay2000** [comment]
I am unable to reproduce the issue. Admittedly, my music library is almost 1/10th of yours. But I have tested with Japanese names (Naruto OST) and they appear to work fine. They should be listed under the globe or three dots group. It would be great if you could get back with following details:
1. What language your phone is set in? (Specific language is not needed. Simply saying English or French would suffice.)
2. Does getting rid of Japanese characters also gets rid of the crash?
3. Can you try connecting to a smaller library and reproduce the crash?
4. If it's not too much to ask, can you try to reproduce the crash with beta [here](http://wpbeta.me/kodi-assist-beta)? This beta has numerous improvements and bug fixes. You'll clearly notice the speed improvements.

**Comment by snz138** [comment]
Actually in reality I have only 686 "artists", but I have not updated the mp3 tags so they aren't up to date with the folder structure. It was not necessary to fix all tags when using foobar2000.
1. My phone is in English.
2. All files (file level) with Japanese are named in romaji (Western characters). Japanese characters exist only in tags. I don't know if this had anything to do with the crash; see below.
3. I reduced the library to contain 109 "artists". XBMC is still reporting over 1232 artists. Cleanup of music library doesn't seem to work.
   It seems I'll need to delete my whole music library inside XBMC and re-create it, but anyway after this the artists list loads up in X.Assist without crashing.
   I didn't have any issues with the library in Android (I maybe used Yatse. I don't remember and it doesn't matter :)
   I'll need to check the behavior again after I've re-created the music library; first smaller then the whole library.
4. I joined the beta program a few hours ago and I am waiting for the e-mail from you for the download link. I'm very interested to test the latest beta.
   If you don't mind me listing these here just for your info, but I am looking for mainly two things:
5. The ability to play an "artist" (all songs by an artist)
6. The ability cache the (music) library onto the phone, but maybe this is not absolutely necessary as loading the artists doesn't seem to take really long

**Comment by snz138** [comment]
1. I have re-created my music library with XBMC listing 221 artists. Browsing works fine without crashing.
   If it is fine by you I'd like to test later with the full library again with the release version and the beta to confirm if I still have an issue or if the issue was a corrupted XBMC MyMusic.db.

**Comment by akshay2000** [comment]
Sorry for not getting back sooner. Long day.
1. Since phone is in English, the Japanese characters will most certainly appear under three dots group.
2. XBMC reads the tags. And title from the tag is what is reported to the client (which can be XBMC skin or an app like X. assist). So, file names are not in play here.
3. I suspect that your library was corrupted. Since you have reported that it worked fine with a smaller library, it would be great if you could test it against the whole library - hopefully, without corruption this time.
4. I am afraid you'll have to wait some more. Though, the new version is based on a totally new API and is super fast as compared to the current one.
5. [Done](https://github.com/akshay2000/XBMCRemoteRT/commit/32b061bbf1081ce4fd68de206a39288eb83bf086).
6. Not sure what you mean here. Do you want all the music (the files themselves) to be made available on phone? Or you just want metadata to be cached on the device so that it loads faster?

**Comment by snz138** [comment]
No worries. Long days here too.
1. I re-created the whole library and X.Assist still exists itself while loading "artists" in "all music".
   So this is reproducable.
2. I now installed the KODI Assist Beta. Thanks for the download link e-mail.
   For some reason the beta does not start at all. It shows the blue start up screen, but then exists.
   I did not uninstall X.Assists as I assume these are both different apps and do not shares files.
3. Cache. I think I just mean the ability to browse the music (or media) library offline on the phone. Then the app needs to only sync the library contents on connect. So just the library (metadata) offline line and not the actual media (mp3 etc.) files of course.
   I don't know yet how fast the beta shows the library so I don't know if such functionality is even needed now.

1-2. Does the app create a log somewhere which I could upload?

**Comment by akshay2000** [comment]
1. Probably Silverlight issue. If the beta solves your problem, I'll close this issue. I'm sorry, but it's not worth supporting the older version.
   2 .That exit is weird. Both apps are different and should not have any clashes at all. Try uninstalling and reinstalling the beta. That should do the trick. That's one weird issue - probably with the device.
2. Again, beta should take care of this. You won't notice lag.

Sorry, no logs. Even if we did have logs, there isn't a way for you to access app's isolated storage directly.

**Comment by snz138** [comment]
1. Yeah, no need to worry about the old app. I think it is best to close this issue and concentrate on getting the new version working.
   I haven't had any issues with any apps I've tried. The phone is Lumia 520 with Windows Phone 8.1. The only apps I've installed are various XBMC apps I've tried and the Facebook app. I don't even have SIM installed on the phone as I am still preferring my older phone.
   I uninstalled (also) XBMC X.Assist and KODI X.Assist. Then re-installed the beta, but it still refuses to start up.
2. Sounds great.

Shall I open an issue on the beta start up issue? I hope there is some way we can debug it to improve this app.

**Comment by akshay2000** [comment]
I went through rigorous testing though different 512 MB devices. The issue could not be reproduced. I suspect some stray data on your device. How about playing around with installation? Like uninstall, restart device, install again. Since the app isn't even starting, I can't even produce a log.

**Comment by snz138** [comment]
I tried this:
Uninstalling KODI Assist Beta.
Shutdown the phone.
Started the phone.
Installed KODI Assist Beta.

But the app still refuses to start. Just exits immediately.
Very strange indeed. I have 4.21 GB free space according to Storage Sense so it shouldn't be about that either. While testing I of course don't have any apps running so 512 MB should be ok.
I also cleared "Temporary files" just in case.
1. Does it help you if I go to "settings / feedback" and set it to ON? Currently it is disabled.
2. As a last resort I could of course reset the phone to factory settings and see if it changes anything.

**Comment by akshay2000** [comment]
I have no idea whether turning on feedback should help. But second options is nuclear. Wouldn't recommend that just for the app. (Although, you're free to try.)  
Before that, one last thing. Where is the app installed, SD card or phone memory? Try switching the storage locations. If that solves it, we have something we can work with.

**Comment by snz138** [comment]
About "feedback": http://support.microsoft.com/kb/2859130
I don't know if this applies to you. I don't know if you can get those logs if I set it ON. What do you think?

A good idea to try. I don't have SD card installed so the app was installed on phone memory. I'll try with an SD card as well.

**Comment by snz138** [comment]
I uninstalled KODI Assist Beta.
Turned off phone.
Tuned on phone.
Installed KODI Assist Beta on SD card.
Still it refuses to start. App closes after start up screen.

I moved the app back to the phone memory and enabled feedback in settings. I started the app a few times trying to read what it says right as it closes, but it is too quick for me to make out the text. It is something "xxxxxxxxxing...", but I can't say what it is.
Anyway the phone says in feedback that it has sent 0,1 KB feedback so it went somewhere.

**Comment by akshay2000** [comment]
Strange. I haven't put up anything like xxxing... in the app start up. Can you bring up the app using Cortana? (Click 'see all' on Cortana.) I'm thinking of adding some sort of logging. So, we'll see if that helps.

**Comment by snz138** [comment]
I could finally read what it says right before it exits Kodi Assist Beta!
When starting the app the splash screen appears. Then it goes away and the screen says shortly "Resuming...".
Does this give you any idea what's going on? Maybe after the first crash it doesn't close properly so WP is actually trying to resume the app. So maybe this doesn't help much.

Cortana: I got it installed on the phone, but I don't understand what to do with it. I don't see "see all" in Cortana. I searched "Kodi Assist" with cortana and it was listed under "Phone". I clicked on it and it showed the app splash screen. Then the screen went black. When I press the "left arrow" to see the open apps I can switch between "Search" (Cortana) and "Kodi Assist Beta". Kodi Assist Beta is just a black screen with no options.
Is this what you mean by bringing the app up with Cortana?
